packages/client/src/components/LogEntryModal/LogEntryModal.tsx

```diff
@@ -70,39 +76,65 @@ const ButtonContainer = styled.div`
   }
 `;
 
-export function CreateLogEntryModal({
+export function LogEntryModal({
+  header,
+  initialValues,
   handleClose,
-  handleCreate,
-}: CreateLogEntryProps) {
+  handleSubmit,
+}: LogEntryModalProps) {
   return (
     <Modal>
       <ModalContent>
         <CloseButton type="button" onClick={handleClose}>
           X
         </CloseButton>
-        <Header>Create Log Entry</Header>
+        <Header>{header}</Header>
         <StyledForm
           onSubmit={(event: React.SyntheticEvent) => {
             event.preventDefault();
             const target = event.target as typeof event.target & {
               logDate: { value: string };
               logValue: { value: string };
             };
-            const logEntry = {
+            let logEntry: LogEntryRequest | EditLogEntryRequest = {
               logDate: new Date(target.logDate.value),
               logValue: parseInt(target.logValue.value, 10),
             };
-            handleCreate(logEntry);
+            // Type guard to check if initialValues is an edit request
+            if (
+              initialValues &&
+              'id' in initialValues &&
+              'logId' in initialValues
+            ) {
+              logEntry = {
+                ...logEntry,
+                id: initialValues.id,
+                logId: initialValues.logId,
+              };
+            }
+            handleSubmit(logEntry);
           }}
         >
           <label htmlFor="logDate">
             Date:&nbsp;
-            <input type="date" name="logDate" />
+            <input
+              type="date"
+              name="logDate"
+              defaultValue={
+                initialValues?.logDate
+                  ? toLocalYYYYMMDD(initialValues.logDate)
```

Comment:
I believe this could be the reason for the bug on save seeming different in the client vs the db?